### PR TITLE
feat: 🎸 get assets held by account

### PR DESCRIPTION
**Pull request by sansan**
### Description

- updates the types for SQ
- adds method to get hel assets

### JIRA Link

https://polymesh.atlassian.net/browse/DA-1035

### Checklist

- [x] Updated the Readme.md (if required) ?

### CODE REVIEW

File: src/middleware/queries.ts [hunk 1]
```diff
@@ -1604,3 +1605,33 @@ export function customClaimTypeQuery(
     variables: { size: size?.toNumber(), start: start?.toNumber(), dids },
   };
 }
+
+/**
+ * @hidden
+ *
+ * Get Confidential Assets hel by a ConfidentialAccount
```
Comment (polymath-eric) [inline src/middleware/queries.ts:1612]:
```suggestion
 * Get Confidential Assets held by a ConfidentialAccount
```

File: src/api/entities/confidential/ConfidentialAccount/index.ts [hunk 2]
```diff
@@ -87,4 +90,24 @@ export class ConfidentialAccount extends Entity<UniqueIdentifiers, string> {
...
     return this.publicKey;
   }
+
+  /**
+   * Retrieve the ConfidentialAssets associated to this Account
+   *
+   * @note uses the middlewareV2
+   * @note there is a possibility that the data is not ready by the time it is requested. In that case, `null` is returned
+   */
+  public async getHeldAssets(): Promise<string[]> {
```
Comment (polymath-eric) [inline src/api/entities/confidential/ConfidentialAccount/index.ts:100]:
```suggestion
  public async getHeldAssets(
    opts: {
      order?: AssetHoldersOrderBy;
      size?: BigNumber;
      start?: BigNumber;
    } = {}
  ): Promise<ResultSet<ConfidentialAsset>> {
```

looking at the Identity signature, I think we should allow pagination options + return the ConfidentialAsset entity

**Review by polymath-eric — CHANGES_REQUESTED** [review]

File: src/middleware/queries.ts [hunk 3]
```diff
@@ -1604,3 +1605,33 @@ export function customClaimTypeQuery(
...
+export function confidentialAssetsByHolderQuery({
+  accountId,
+}: QueryArgs<ConfidentialAssetHolder, 'accountId'>): QueryOptions<
+  QueryArgs<ConfidentialAssetHolder, 'accountId'>
+> {
+  const query = gql`
+    query ConfidentialAssetsByAccount($accountId: String!) {
+      confidentialAssetHolders(
+        filter: { accountId: { equalTo: $accountId } }
+        orderBy: [${TickerExternalAgentHistoriesOrderBy.CreatedAtAsc}, ${TickerExternalAgentHistoriesOrderBy.CreatedBlockIdAsc}]
```
Comment (prashantasdeveloper) [inline src/middleware/queries.ts:1623]: It seems this got left over from copy paste

**Review by prashantasdeveloper — CHANGES_REQUESTED** [review]

File: src/api/entities/confidential/ConfidentialAccount/index.ts [hunk 4]
```diff
@@ -198,6 +202,7 @@ export class ConfidentialAccount extends Entity<UniqueIdentifiers, string> {
 
   /**
    * Determine whether this Account exists on chain
+   *
```
Comment (polymath-eric) [inline src/api/entities/confidential/ConfidentialAccount/index.ts:205]:
```suggestion
```

File: src/api/entities/confidential/__tests__/ConfidentialAccount/index.ts [hunk 5]
```diff
@@ -228,4 +229,39 @@ describe('ConfidentialAccount class', () => {
       return expect(account.exists()).resolves.toBe(false);
     });
   });
+
+  describe('method: getHeldAssets', () => {
+    it('should return an array of ConfidentialAssets held by the ConfidentialAccount', async () => {
+      const assetId = '76702175d8cbe3a55a19734433351e25';
+      const asset = new ConfidentialAsset({ id: assetId }, context);
```
Comment (polymath-eric) [inline src/api/entities/confidential/__tests__/ConfidentialAccount/index.ts:236]: I think SDK uses `entityMockUtils.getConfidentialAsset()` type methods, but seems like a real one works here.

**Review by polymath-eric — APPROVED** [review]

**Comment by polymath-eric** [comment]
left 2 nits, but otherwise lgtm

**Pull request merged**